samples/clip-geometry/src/main/java/com/esri/arcgismaps/sample/clipgeometry/components/ClipGeometryViewModel.kt

```diff
@@ -0,0 +1,177 @@
+/* Copyright 2024 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.clipgeometry.components
+
+import android.app.Application
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.arcgismaps.Color
+import com.arcgismaps.geometry.Envelope
+import com.arcgismaps.geometry.Geometry
+import com.arcgismaps.geometry.GeometryEngine
+import com.arcgismaps.geometry.Point
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.symbology.SimpleFillSymbol
+import com.arcgismaps.mapping.symbology.SimpleFillSymbolStyle
+import com.arcgismaps.mapping.symbology.SimpleLineSymbol
+import com.arcgismaps.mapping.symbology.SimpleLineSymbolStyle
+import com.arcgismaps.mapping.view.Graphic
+import com.arcgismaps.mapping.view.GraphicsOverlay
+import com.arcgismaps.toolkit.geoviewcompose.MapViewProxy
+import com.esri.arcgismaps.sample.clipgeometry.R
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import kotlinx.coroutines.launch
+
+class ClipGeometryViewModel(application: Application) : AndroidViewModel(application) {
+    // create  a map with a topographic basemap style
+    val arcGISMap = ArcGISMap(BasemapStyle.ArcGISTopographic)
+    // create a MapViewProxy to interact with the MapView
+    val mapViewProxy = MapViewProxy()
+
+    // graphics overlay to display graphics
+    val graphicsOverlay = GraphicsOverlay()
```

Comment:
```suggestion
        val graphicsOverlay = GraphicsOverlay().apply {
            graphics.add(coloradoGraphic)
            graphics.addAll(envelopesGraphics)
    }
```

I'd suggest moving this below declaration of `coloradoGraphic` and `envelopesGraphics` and then  removing the call to `reset()` in the init block

Reply:
As a broad suggestion--the less work you can do in the init block the better. Keeps code cleaner and more maintainable since init runs after all declarations

Reply:
I wasn't a fan of calling `reset()` in the init block so this is much better